internal/db/db.go

```diff
@@ -21,11 +21,13 @@ type dB struct {
 }
 
 type ComposeEntry struct {
-	Id        uuid.UUID
-	Request   json.RawMessage
-	CreatedAt time.Time
-	ImageName *string
-	ClientId  *string
+	Id               uuid.UUID
+	BlueprintId      uuid.UUID
+	BlueprintVersion int
+	Request          json.RawMessage
+	CreatedAt        time.Time
+	ImageName        *string
+	ClientId         *string
 }
```

Comment:
I am not a huge fan of adding fields from joined tables into the "entry" types. That being said, I am not that experienced yet with how this should be handled in Go. In my other project, I use embedding approach for result-only types:

```
type ComposeVersion struct {
	Compose
	Version
}
```

Then you simply:

```
cv := ComposeVersion{}
err = result.Scan(&cv.Compose.field1, &cv.Compose.field2, ...)
```

Embedded fields can be actually accessed directly: `&cv.field1` as long as they are unique. For `id` you will have to use the "long" form.

It also maybe makes sense to figure out a naming convention which would indicate this type is not an "entry" by not having `Entry` suffix. Meaning: you cannot insert it, this is just a result (of a join).

In fact, in my other project I use the Scanny library where you define Go struct tags and the library does the mapping automatically:

```
type ComposeVersion struct {
	Compose    `db:"c"`
	Version `db:"v"`
}
```

Then you do this:

```
result := &model.ComposeVersion{}
pgxscan.Get(ctx, Pool, result, "SELECT ...")

Unfortunately, postgres never returns column names with prefixes so these must be entered explicitly one by one:

```
SELECT
 c.id as "c.id",
 c.name as "c.name"
 FROM ... join c and v
```

This is only ugly for these joins, other than that Scanny is great.